npc/namespace.go

```diff
@@ -471,7 +471,19 @@ func (ns *ns) deleteNamespace(obj *coreapi.Namespace) error {
 	}
 
 	// Remove namespace ipset from any matching namespace selectors
-	return ns.nsSelectors.delFromMatching(obj.ObjectMeta.UID, obj.ObjectMeta.Labels, string(ns.allPods.ipsetName))
+	err := ns.nsSelectors.delFromMatching(obj.ObjectMeta.UID, obj.ObjectMeta.Labels, string(ns.allPods.ipsetName))
+	if err != nil {
+		return err
+	}
+
+	// Flush defaultAllowIPSet
```

Comment:
`defaultAllowIPSet` is created only in the non-legacy mode, so you need to wrap the flush and destroy ops with `if !ns.legacy { ... }`.

Reply:
Done, pushing the ammended commit..